scripts/scraper/process-macros/extract-macro-calls.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Given the name of a macro and some HTML,
+ * find the all invocations of the macro in the HTML.
+ *
+ * Return an array, containing one element per invocation,
+ * where each element is itself an array containing the arguments
+ * with which the macro was called in that invocation.
+ */
+function extractMacroCalls(macroName, html) {
+  const regex = new RegExp(`{{\\ ?${macroName}\\((.*?)\\)}}`, 'gi');
```

Comment:
Don't ask me why I noticed this but if this is the input:
```html
<p>{{EmbedLiveSample("Examples")}}</p>

<div>{{ EmbedLiveSample("Examples2")}}</div>

<pre>{{EmbedLiveSample('Examples3') }}</pre>
```
...it would only return `[ [ 'Examples' ], [ 'Examples2' ] ]`

Reply:
By the way, the `String.matchAll` is neat. Try
```javascript
function extractMacroCalls(macroName, html) {
    const regex = new RegExp(`{{\\s*${macroName}\\(["'](.*)["']\\)\\s*}}`, "g");
    return [...html.matchAll(regex)].map(m => m[1]);
}
```

Reply:
`matchAll` would be great here but [it's only supported from Node 12](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#Browser_compatibility). I understood from https://github.com/mdn/stumptown-content/issues/141 that we want to support the current LTS release (which seems sensible), and that's v10.

Reply:
> if this is the input...it would only return [ [ 'Examples' ], [ 'Examples2' ] ]

Ugh. I've pushed an update that seems to fix this particular regex bug.

Reply:
Good catch. I didn't know this so now I wonder where else I've foolishly added that.

Reply:
Mind you, my little suggestion had two other tricks that might be worth to keep. I.e. `\\s*` instead of `\\ ?` so it can match 0 or many. And the `["'](.*)["']` should mean you don't need to clean up the match in the iterator. But perhaps I've misunderstood something about how these macros always work.

Reply:
I'm always happy to accept help with regexes as my understanding of them is a bit primitive. But this:

```js
function extractMacroCalls(macroName, html) {
    const regex = new RegExp(`{{\\s*${macroName}\\(["'](.*)["']\\)\\s*}}`, "g");
    return [...html.matchAll(regex)].map(m => m[1]);
}
```

does not seem to work, even with `matchAll` support:

```
> var s = `<p>{{EmbedLiveSample("Examples")}}</p><div>{{ EmbedLiveSample("Examples2")}}</div><pre>{{EmbedLiveSample('Examples3') }}</pre>`
> extractMacroCalls('EmbedLiveSample', s)
< Array [ "Examples\")}}</p><div>{{ EmbedLiveSample(\"Examples2\")}}</div><pre>{{EmbedLiveSample('Examples3" ]
```